src/prefect/settings/models/server/database.py

```diff
@@ -15,6 +15,42 @@
 from prefect.settings.base import PrefectBaseSettings, build_settings_config
 
 
+class SQLAlchemyTLSSettings(PrefectBaseSettings):
+    """
+    Settings for controlling SQLAlchemy mTLS context when
+    using a PostgreSQL database.
+    """
+
+    model_config: ClassVar[SettingsConfigDict] = build_settings_config(
+        ("server", "database", "sqlalchemy", "connect_args", "tls")
+    )
+
+    enabled: bool = Field(
+        default=False,
+        description="Controls whether connected to mTLS enabled PostgreSQL when using a PostgreSQL database with the Prefect backend.",
+    )
+
+    ca_file: str = Field(
+        default="",
+        description="This configuration settings option specifies the path to PostgreSQL client certificate authority file.",
+    )
+
+    cert_file: str = Field(
+        default="",
+        description="This configuration settings option specifies the path to PostgreSQL client certificate file.",
+    )
+
+    key_file: str = Field(
+        default="",
+        description="This configuration settings option specifies the path to PostgreSQL client key file.",
+    )
```

Comment:
i think these fields should be `Optional` and default to `None` to disambiguate the "not set" case

Reply:
sure, I will add more validation to ensure it passed static check

Reply:
@zzstoatzz  please help to verify.

1. when root ca not provided, use system default
2. when client key and client cert provided, load certificate chain